lib/AST/ASTImporter.cpp

```diff
@@ -71,6 +71,25 @@
 #include <type_traits>
 #include <utility>
 
+namespace {
+
+// FIXME Do we have something similar in LLVM?
+template <typename Function> class Finally {
+public:
+  Finally(const Function &function) : function(function) {}
+  ~Finally() { function(); }
+
+private:
+  Function function;
+};
+
+template <typename Function>
+Finally<Function> finally(const Function &function) {
+  return Finally<Function>(function);
+}
+
+} // namespace unnamed
+
```

Comment:
Until this is only used for `ImportPath` push and pop there may be simpler to make a class that does push in constructor and pop in destructor.